src/main/java/Duke.java

```diff
@@ -1,10 +1,112 @@
+                }
+
+                switch (command) {
+                    case LIST: {
+                        System.out.println(BAR);
+                        System.out.println(INDENTATION + "list");
+                        tasks.printAll();
+                        System.out.println(BAR);
+                        break;
+                    }
+                    case BYE: {
+                        echo("Bye. Hope to see you again soon!");
+                        scanner.close();
+                        break;
+                    }
+                    case DEADLINE: {
+                        String description = commandArgs[0];
+                        String by = commandArgs[1];
+                        Deadline deadline = new Deadline(description, by);
+                        tasks.addTask(deadline);
+                        int numTasks = tasks.getNumTasks();
+                        echo("Got it. I've added this task:", "  " + deadline.toString(),
+                                "Now you have " + numTasks + " tasks in the list.");
+                        break;
+                    }
+                    case EVENT: {
+                        String description = commandArgs[0];
+                        String from = commandArgs[1];
+                        String to = commandArgs[2];
+                        Event event = new Event(description, from, to);
+                        tasks.addTask(event);
+                        int numTasks = tasks.getNumTasks();
+                        echo("Got it. I've added this task:", "  " + event.toString(),
+                                "Now you have " + numTasks + " tasks in the list.");
+                        break;
+                    }
+                    case TODO: {
+                        String description = commandArgs[0];
+                        Task task = new ToDo(description);
+                        tasks.addTask(task);
+                        int numTasks = tasks.getNumTasks();
+                        echo("Got it. I've added this task:", "  " + task.toString(),
+                                "Now you have " + numTasks + " tasks in the list.");
+                        break;
+                    }
+                    case MARK: {
+                        int taskIndex = Integer.parseInt(commandArgs[0]);
+                        echo("Nice! I've marked this task as done:",
+                                " " + tasks.markTask(taskIndex));
+                        break;
+                    }
+                    case UNMARK: {
+                        int taskIndex = Integer.parseInt(commandArgs[0]);
+                        echo("OK, I've marked this task as not done yet:",
+                                " " + tasks.unmarkTask(taskIndex));
+                        break;
+                    }
+                }
+                // System.out.println("Success! Your command is " + command.text);
+            } catch (Error e) {
```

Comment:
Should you try to catch some other (and more specific) Exceptions instead? Maybe create a custom one? Errors generally indicate serious problems and it's probably [best not to attempt to catch them](https://docs.oracle.com/javase/7/docs/api/java/lang/Error.html).